responses.py

```diff
@@ -178,12 +183,14 @@ def get_headers(self):
     def get_response(self, request):
         raise NotImplementedError
 
-    def matches(self, request):
+    def matches(self, request, url_object_equality=False):
```

Comment:
if we need this here, per below, we should have it be a pure equality check (e.g. ``__eq__``), but if its going to rename on this method it should be something more generic like ``strict``.